drivers/watchdog/wdt_dw.c

```diff
@@ -15,14 +15,38 @@
 #include "wdt_dw.h"
 #include "wdt_dw_common.h"
 
+/* Check if reset property is defined */
+#define WDT_DW_IS_INST_RESET_EN(inst)	DT_NODE_HAS_PROP(DT_DRV_INST(inst), resets)
+#define WDT_DW_CHECK_RESET(inst)	WDT_DW_IS_INST_RESET_EN(inst) ||
+#define WDT_DW_RESET_SUPPORT		(DT_INST_FOREACH_STATUS_OKAY(WDT_DW_CHECK_RESET) 0)
```

Comment:
I'm confused by this. You are checking if a reset line exists in one of the instances in the DT? What about the other instances that do not have it?